aws/rust-runtime/aws-inlineable/src/presigning.rs

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+/// Presigning config and builder
```

Comment:
can you add warnings somewhere that will be really easy to find telling people that `expires_in` is superceded by expiring credentials and if the credentials expire before the expiry, the link will stop working? I know a lot of other SDKs have gotten bugs from this detail